core/src/utils/content/content.utils.spec.ts

```diff
@@ -0,0 +1,189 @@
+        }, 0);
+
+        expect(scrollToMock).toHaveBeenCalledWith({
+          top: 0,
+          left: 0,
+          behavior: 'auto'
+        });
+      });
+
+    });
+
+    describe('scroll duration is greater than 0', () => {
+
+      it('should smooth scroll ion-content', () => {
+        const scrollToTopMock = jest.fn();
+
+        scrollToTop(<any>{
+          tagName: 'ION-CONTENT',
+          scrollToTop: scrollToTopMock
+        }, 300);
+
+        expect(scrollToTopMock).toHaveBeenCalledWith(300);
+      });
+
+      it('should smooth scroll the element', async () => {
+        const scrollToMock = jest.fn();
+
+        await scrollToTop(<any>{
+          tagName: 'DIV',
+          scrollTo: scrollToMock
+        }, 300);
+
+        expect(scrollToMock).toHaveBeenCalledWith({
+          top: 0,
+          left: 0,
+          behavior: 'smooth'
+        });
+      });
+
+    });
+
+  });
+
+  describe('scrollByPoint', () => {
+
+    describe('scroll duration is 0', () => {
+
+      it('should call scrollByPoint when the tag name is ion-content', async () => {
+        const scrollByPointMock = jest.fn();
+
+        await scrollByPoint(<any>{
+          tagName: 'ION-CONTENT',
+          scrollByPoint: scrollByPointMock
+        }, 10, 15, 0);
+
+        expect(scrollByPointMock).toHaveBeenCalledWith(10, 15, 0);
+      });
+
+      it('should call the element scrollBy when the tag name is not ion-content', async () => {
+        const scrollByMock = jest.fn();
+
+        await scrollByPoint(<any>{
+          tagName: 'DIV',
+          scrollBy: scrollByMock
+        }, 10, 15, 0);
+
+        expect(scrollByMock).toHaveBeenCalledWith({
+          top: 15,
+          left: 10,
+          behavior: 'auto'
+        });
+      });
+
+    });
+
+    describe('scroll duration is greater than 0', () => {
+
+      it('should smooth scroll ion-content', async () => {
+        const scrollByPointMock = jest.fn();
+
+        await scrollByPoint(<any>{
+          tagName: 'ION-CONTENT',
+          scrollByPoint: scrollByPointMock
+        }, 10, 15, 300);
+
+        expect(scrollByPointMock).toHaveBeenCalledWith(10, 15, 300);
+      });
+
+      it('should smooth scroll the element', async () => {
+        const scrollByMock = jest.fn();
+
+        await scrollByPoint(<any>{
+          tagName: 'DIV',
+          scrollBy: scrollByMock
+        }, 10, 15, 300);
+
+        expect(scrollByMock).toHaveBeenCalledWith({
+          top: 15,
+          left: 10,
+          behavior: 'smooth'
+        });
+      });
+
+    });
+
+  });
+
+  it('printIonContentErrorMsg should display "<my-el> must be used inside ion-content or .ion-content-scroll-host."', () => {
+    const consoleErrorMock = jest.spyOn(console, 'error').mockImplementation();
+
+    printIonContentErrorMsg(<any>{
+      tagName: 'MY-EL'
+    });
+
+    expect(consoleErrorMock).toHaveBeenCalledWith('<my-el> must be used inside ion-content or .ion-content-scroll-host.');
```

Comment:
This is confusing. If I use `<my-el>` inside of `.ion-content-scroll-host`, I would expect the component to work as intended. The reality is that the component needs to be inside of `ion-content`. Whether or not you use `.ion-content-scroll-host` depends on if you are using virtual scroll.